pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarConnectionFactory.java

```
@@ -1894,6 +1914,26 @@ public synchronized int getConnectionConsumerStopTimeout() {
     return connectionConsumerStopTimeout;
   }
 
+  public void closeTemporaryProducerForDestination(
+      PulsarDestination defaultDestination, boolean transactions) throws JMSException {
+    String fullyQualifiedTopicName = getPulsarTopicName(defaultDestination);
+    String key = computeKeyFromTopicName(transactions, fullyQualifiedTopicName);
+    tempProducers.computeIfPresent(
+        key,
+        (k, tempProducer) -> {
+          try {
+            tempProducer.close();
+          } catch (Exception e) {
+            log.error(
+                "Failed to close temporary Producer for topic: {}",
```

we should check if closing a pulasr producer twice leads to an error, in that case we can swallow it

---

Closing a producer twice is not leading to an error. Tested it with temporary code change and reverted it.